src/main/java/io/cdap/plugin/sap/SapODataSchemas.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.sap;
+
+import io.cdap.cdap.api.data.schema.Schema;
+
+/**
+ * Utility methods to construct SAP OData schemas.
+ */
+public class SapODataSchemas {
+
+  private SapODataSchemas() {
+    throw new AssertionError("Should not instantiate static utility class.");
+  }
+
+  public static Schema streamSchema(String name) {
+    return Schema.recordOf(
+      name + "-stream-record",
+      Schema.Field.of(SapODataConstants.STREAM_ETAG_FIELD_NAME, Schema.nullableOf(Schema.of(Schema.Type.STRING))),
+      Schema.Field.of(SapODataConstants.STREAM_CONTENT_TYPE_FIELD_NAME,
+                      Schema.nullableOf(Schema.of(Schema.Type.STRING))),
+      Schema.Field.of(SapODataConstants.STREAM_READ_LINK_FIELD_NAME, Schema.nullableOf(Schema.of(Schema.Type.STRING))),
+      Schema.Field.of(SapODataConstants.STREAM_EDIT_LINK_FIELD_NAME, Schema.nullableOf(Schema.of(Schema.Type.STRING))));
+  }
+
+  public static Schema pointSchema(String name) {
```

Comment:
why does it need to generate a name instead of using a constant value? Isn't the schema always representing the same thing?

Reply:
Changed to be a constant. 

I think I faced issues with records' names overlapping during development(if record schema with name 'test' contains a field with a record schema with the same name `test`).
Anyway, there are no such issues with the current mapping, so schemas changed to be a constant. 

Thanks.